homeassistant/components/solax/sensor.py

```diff
@@ -79,8 +81,9 @@ def __init__(self, hass, api):
 class Inverter(Entity):
     """Class for a sensor."""
 
-    def __init__(self, key, unit):
+    def __init__(self, inverter_name, key, unit):
```

Comment:
This seems wrong. The serial is not the name ? 

And you need to return the serial from the `unique_id` property.

Reply:
Surely not just the serial number for ‘unique_id’? Then all the sensors for one inverter would return the same ‘unique_id’. I’ll have to find an example in the docs.

Reply:
@balloob is the entity_id value distinct from the unique_id?

Reply:
Ok, I think I understand `unique_id` better now. Using the serial number of the inverter and the index of the sensor.